app/workers/deployment_watchdog.rb

```diff
@@ -0,0 +1,21 @@
+class DeploymentWatchdogWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: 'deployment_watchdog'
+  def perform(auto_deployment_id)
+    auto_deployment = AutoDeployment.find_one(id: auto_deployment_id)
+    # an inactive auto_deployment was either deployed, or superceded
+    # by another auto_deployment. Either way we exit without notifying.
+    return logger.debug "auto_deployment id #{auto_deployment.id} is inactive which means it was already deployed or superceded." unless auto_deployment.active?
+    if auto_deployment.pending?
+      logger.info "There was an issue with your deployment. Seems to be hung on #{auto_deployment.pending_statuses}"
+      message_class = AutoDeploymentStuckPendingMessage
+    elsif auto_deployment.failed? 
```

Comment:
This case is already handled by SlashDeploy:

https://github.com/remind101/slashdeploy/blob/adea526cbfac35f7b001ae2984bc4f382299800e/spec/features/auto_deploy_spec.rb#L262-L265